gluten-core/src/main/java/org/apache/gluten/vectorized/JniLibLoader.java

```diff
@@ -159,10 +160,16 @@ public static void unloadFromPath(String libPath) {
   }
 
   public void mapAndLoad(String unmappedLibName, boolean requireUnload) {
+    mapAndLoad(Optional.empty(), unmappedLibName, requireUnload);
+  }
+
+  public void mapAndLoad(
+      Optional<String> packagePath, String unmappedLibName, boolean requireUnload) {
```

Comment:
Would suggest keeping one single `mapAndLoad` then doing the concatenation in caller code.

Probably just rename the arg in previous `mapAndLoad` method from `unmappedLibName` to `unmappedLibPath`?

Reply:
Makes sense, I will do that

Reply:
Can the `System.mapLibraryName(unmappedLibName)` inside `mapAndLoad` method be moved to the calling method?

Reply:
> Can the `System.mapLibraryName(unmappedLibName)` inside `mapAndLoad` method be moved to the calling method?

Sounds good to me.

Reply:
Changed, thank you

Reply:
And can we do the same on `loadAndCreateLink` too? For consistency. Though I assume that one requires for some refactors in caller code.